commonmark-ext-image-attributes/src/main/java/org/commonmark/ext/image/attributes/internal/ImageAttributesDelimiterProcessor.java

```diff
@@ -0,0 +1,131 @@
+package org.commonmark.ext.image.attributes.internal;
+
+import org.commonmark.ext.image.attributes.ImageAttributes;
+import org.commonmark.node.Image;
+import org.commonmark.node.Node;
+import org.commonmark.node.Text;
+import org.commonmark.parser.delimiter.DelimiterProcessor;
+import org.commonmark.parser.delimiter.DelimiterRun;
+
+import java.util.*;
+
+public class ImageAttributesDelimiterProcessor implements DelimiterProcessor {
+
+    // Only allow a defined set of attributes to be used.
+    private static final Set<String> SUPPORTED_ATTRIBUTES = Collections.unmodifiableSet(
+            new HashSet<>(Arrays.asList("width", "height")));
+
+    @Override
+    public char getOpeningCharacter() {
+        return '{';
+    }
+
+    @Override
+    public char getClosingCharacter() {
+        return '}';
+    }
+
+    @Override
+    public int getMinLength() {
+        return 1;
+    }
+
+    @Override
+    public int getDelimiterUse(DelimiterRun opener, DelimiterRun closer) {
+        return 1;
+    }
+
+    @Override
+    public void process(Text opener, Text closer, int delimiterCount) {
+        // Check if the attributes can be applied - if the previous node is an Image, and if all the attributes are in
+        // the set of SUPPORTED_ATTRIBUTES
+        if (opener.getPrevious() instanceof Image) {
+            boolean canApply = true;
+            List<Node> toUnlink = new ArrayList<>();
+
+            StringBuilder attributesString = new StringBuilder();
+            Node tmp = opener.getNext();
+            while (tmp != null && tmp != closer) {
+                Node next = tmp.getNext();
+                // Only Text nodes can be used for attributes
+                if (tmp instanceof Text) {
+                    String attributes = ((Text) tmp).getLiteral();
+                    for (String s : attributes.split("\\s+")) {
+                        String[] attribute = s.split("=");
+                        // Check if the attribute is in SUPPORTED_STYLES.
```

Comment:
```suggestion
                        // Check if the attribute is in SUPPORTED_ATTRIBUTES.
```

Reply:
Though... not really sure if the comment is necessary.

Reply:
I think you're right that it's not necessary, so I've removed it.